apps/api/src/database/schema/voting/defs.ts

```diff
@@ -27,96 +28,156 @@ export const epochs = pgTable("epochs", {
   amount: tokenAmount("amount").notNull(),
   address: blockchainAddress("address").notNull(),
   expiresAt: timestamp("expires_at", { withTimezone: true }).notNull(),
   withdrawalAt: timestamp("withdrawal_at", { withTimezone: true }),
   withdrawnAt: timestamp("withdrawn_at", { withTimezone: true }),
   epochCreated: uuid("epoch_created").references(() => epochs.id),
-  createdAt: timestamp("created_at", { withTimezone: true }).defaultNow().notNull(),
-  updatedAt: timestamp("updated_at", { withTimezone: true }).defaultNow().notNull(),
+  createdAt: timestamp("created_at", { withTimezone: true })
+    .defaultNow()
+    .notNull(),
+  updatedAt: timestamp("updated_at", { withTimezone: true })
+    .defaultNow()
+    .notNull(),
 });
 
-export const voteAssignments = pgTable("vote_assignments", {
-  stakeId: uuid("stake_id").notNull().references(() => stakes.id, { onDelete: "cascade" }),
-  userId: uuid("user_id").notNull().references(() => users.id, { onDelete: "cascade" }),
-  epoch: uuid("epoch").notNull().references(() => epochs.id, { onDelete: "cascade" }),
-  createdAt: timestamp("created_at", { withTimezone: true }).defaultNow().notNull(),
-  amount: tokenAmount("amount").notNull(),
-}, (table) => [
-  index("idx_vote_assignments_user_id").on(table.userId),
-  index("idx_vote_assignments_epoch").on(table.epoch),
-  index("idx_vote_assignments_stake_id").on(table.stakeId),
-]);
+export const voteAssignments = pgTable(
+  "vote_assignments",
+  {
+    stakeId: uuid("stake_id")
+      .notNull()
+      .references(() => stakes.id, { onDelete: "cascade" }),
+    userId: uuid("user_id")
+      .notNull()
+      .references(() => users.id, { onDelete: "cascade" }),
+    epoch: uuid("epoch")
+      .notNull()
+      .references(() => epochs.id, { onDelete: "cascade" }),
+    createdAt: timestamp("created_at", { withTimezone: true })
+      .defaultNow()
+      .notNull(),
+    amount: tokenAmount("amount").notNull(),
+  },
+  (table) => [
+    index("idx_vote_assignments_user_id").on(table.userId),
+    index("idx_vote_assignments_epoch").on(table.epoch),
+    index("idx_vote_assignments_stake_id").on(table.stakeId),
+  ],
+);
 
-export const votesAvailable = pgTable("votes_available", {
-  userId: uuid("user_id").notNull().references(() => users.id, { onDelete: "cascade" }),
-  epoch: uuid("epoch").notNull().references(() => epochs.id, { onDelete: "cascade" }),
-  amount: tokenAmount("amount").notNull(),
-  createdAt: timestamp("created_at", { withTimezone: true }).defaultNow().notNull(),
-  updatedAt: timestamp("updated_at", { withTimezone: true }).defaultNow().notNull(),
-}, (table) => [
-  primaryKey({ columns: [table.userId, table.epoch], name: "votes_available_pkey" }),
-  index("idx_votes_available_user_id").on(table.userId),
-]);
+export const votesAvailable = pgTable(
+  "votes_available",
+  {
+    userId: uuid("user_id")
+      .notNull()
+      .references(() => users.id, { onDelete: "cascade" }),
+    epoch: uuid("epoch")
```

Comment:
I thought votes available would be a global thing and not really tied to a particular epoch. As far as I understand, you can vote on epoch 2 using the votes from epoch 1.

Reply:
That was my understanding as well, until few days ago.